src/key_gen/mod.rs

```diff
@@ -1048,7 +1084,7 @@ impl KeyGen {
             initalization_accumulator: InitializationAccumulator::new(),
             complaints_accumulator: ComplaintsAccumulator::new(names, threshold),
             pending_complain_messages: Vec::new(),
-            pending_messages: Vec::new(),
+            caching_messages: BTreeMap::new(),
```

Comment:
```suggestion
            cached_sent_messages: BTreeMap::new(),
```
The more I'm reading through the more i think this needs a rename for clarity

Reply:
I am not at naming stuff.
Any suggestion?

Reply:
I think `message_cache` sounds reasonable.